src/pods_distribution.py

```diff
@@ -0,0 +1,180 @@
+
+    def __init__(self, container_dir: str):
+        self.container_dir = container_dir
+        self.id = os.path.basename(container_dir)
+
+        config = self._get_config()
+        self.container_type = dpath.util.get(config, "annotations/io.kubernetes.cri.container-type")
+        self.container_name = dpath.util.get(config, "annotations/io.kubernetes.cri.container-name", default=None)
+        self.pod_name = dpath.util.get(config, "annotations/io.kubernetes.cri.sandbox-name")
+        self.pod_namespace = dpath.util.get(config, "annotations/io.kubernetes.cri.sandbox-namespace")
+
+        self.disk_size = self._get_disk_size()
+
+    def _get_config(self):
+        config_file_path = os.path.join(self.container_dir, "config.json")
+        with open(config_file_path, "r") as config_file:
+            json_config = config_file.read()
+        return json.loads(json_config)
+
+    def _get_disk_size(self) -> int:
+        root_fs_dir = os.path.join(self.container_dir, "rootfs")
+        return DiskUtils.get_size(root_fs_dir)
+
+    def __repr__(self):
+        return f"Container(container_type={self.container_type}, " \
+               f"container_name={self.container_name}, disk_size={self.disk_size})"
+
+
+class Pod:
+    pod_name: str
+    namespace: str
+
+    def __init__(self, pod_name: str, namespace: str):
+        self.pod_name = pod_name
+        self.namespace = namespace
+
+    def __eq__(self, other):
+        return self.pod_name == other.pod_name and self.namespace == self.namespace
+
+    def __hash__(self):
+        return hash((self.pod_name, self.namespace))
+
+    def __repr__(self):
+        return f"Pod(pod_name={self.pod_name}, namespace={self.namespace})"
+
+
+def get_pods_disk_distribution() -> (Dict[Pod, List[Container]], List[str]):
+    partitions = psutil.disk_partitions(all=True)
+
+    # Create a list of container objects
+    warnings: List[str] = []
+    containers: List[Container] = []
+    for p in partitions:
+        if not p.mountpoint.endswith("/rootfs"):
+            continue
+
+        container_dir = os.path.dirname(p.mountpoint)
+        try:
+            c = Container(container_dir)
+            containers.append(c)
+        except Exception as e:
+            warnings.append(f"failed to get container information from {container_dir}: " + str(e))
+
+    # Create a mapping between pods and the containers they host
+    current_pod = Pod(os.environ["CURRENT_POD_NAME"], os.environ["CURRENT_POD_NAMESPACE"])
+    pods_to_containers: Dict[Pod, List[Container]] = {}
+    for c in containers:
+        pod = Pod(c.pod_name, c.pod_namespace)
+        if pod == current_pod:  # exclude the containers of the pod that is currently executing
+            continue
+
+        if pod not in pods_to_containers and pod != current_pod:
+            pods_to_containers[pod] = []
+        pods_to_containers[pod].append(c)
+
+    return pods_to_containers, warnings
+
+
+class DiskStats:
+    total: int
```

Comment:
Also here.

https://home.robusta.dev/blog/common-python-mistakes-mixing-up-instance-and-class-members/

Reply:
same as above